From 783bde4ffa73a520d20b7e3745b3a7153381f3bc Mon Sep 17 00:00:00 2001
From: Huang Chenyi <b44402@freescale.com>
Date: Wed, 29 May 2013 16:46:38 +0800
Subject: [PATCH 14/39] ENGR00262367 MTP:Other users can't acccess the MTP
 content from PC except owner

Adjust the order of storage volumes in "filtered" list,
keep the primary strorage volume in the head positon.

Signed-off-by: Huang Chenyi <b44402@freescale.com>
---
 services/java/com/android/server/MountService.java | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/services/java/com/android/server/MountService.java b/services/java/com/android/server/MountService.java
index e60231a..7a260be 100644
--- a/services/java/com/android/server/MountService.java
+++ b/services/java/com/android/server/MountService.java
@@ -93,6 +93,8 @@ import java.util.Map;
 import java.util.Map.Entry;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
+import java.util.Collections;
+import java.util.ListIterator;
 
 import javax.crypto.SecretKey;
 import javax.crypto.SecretKeyFactory;
@@ -2256,6 +2258,18 @@ class MountService extends IMountService.Stub
                     filtered.add(volume);
                 }
             }
+            ListIterator<StorageVolume> it = filtered.listIterator();
+            int index = -1;
+            while(it.hasNext()){
+               StorageVolume tmp = filtered.get(++index);
+               if(tmp.isPrimary())
+                     break;
+            }
+            if(index != 0){
+               StorageVolume staging = filtered.remove(index);
+               filtered.add(staging);
+               Collections.rotate(filtered,1);
+            }
             return filtered.toArray(new StorageVolume[filtered.size()]);
         }
     }
-- 
1.8.0

